realm/realm-library/src/main/java/io/realm/log/RealmLog.java

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright 2016 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.log;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.realm.internal.Keep;
+import io.realm.internal.Util;
+
+/**
+ * Global logger used by all Realm components.
+ * Custom loggers can be added by registering classes implementing {@link Logger}.
+ */
+@Keep
+public final class RealmLog {
+
+    private static final Logger[] NO_LOGGERS = new Logger[0];
+
+    // All of the below should be modified together under under a lock on LOGGERS.
+    private static final List<Logger> LOGGERS = new ArrayList<>();
+    private static volatile Logger[] loggersAsArray = NO_LOGGERS;
+    private static volatile int minimumNativeLogLevel = Integer.MAX_VALUE;
```

Comment:
any read and write of `minimumNativeLogLevel` is synchronized with `LOGGERS`.
I think no extra volatile is necessary